incubator/group/keeper.go

```diff
@@ -101,35 +131,34 @@ func NewGroupKeeper(storeKey sdk.StoreKey, paramSpace params.Subspace) Keeper {
 	//
 	// Group Account Table
 	//
-	groupAccountTableBuilder := orm.NewNaturalKeyTableBuilder(GroupAccountTablePrefix, storeKey, &GroupAccountMetadataBase{}, orm.Max255DynamicLengthIndexKeyCodec{})
+	k.groupAccountSeq = orm.NewSequence(storeKey, GroupAccountTableSeqPrefix)
+	groupAccountTableBuilder := orm.NewNaturalKeyTableBuilder(GroupAccountTablePrefix, storeKey, &StdGroupAccountMetadata{}, orm.Max255DynamicLengthIndexKeyCodec{})
 	k.groupAccountByGroupIndex = orm.NewUInt64Index(groupAccountTableBuilder, GroupAccountByGroupIndexPrefix, func(value interface{}) ([]uint64, error) {
-		group := value.(*GroupAccountMetadataBase).Group
+		group := value.(*StdGroupAccountMetadata).Base.Group
```

Comment:
I have some issues to move the code to `GroupAccountMetadataI`. When creating a proposal I calculate the `timeout` currently based on what's defined in the threshold policy. If that information is not available anymore then I can not check it during vote or exec. Any ideas?
I was wondering if we can move this redesign into a new PR so that this one focuses on the happy path workflow only

Reply:
Ideally all of those concerns would be abstracted into the `DecisionPolicy` interface.

But it's fine to move to a separate issue and PR.

Reply:
Added a ticket #48